resources/scripts/download_libs.py

```diff
@@ -181,37 +111,48 @@ def retrieve_libraries_for_specific_version(
     """
     s3 = boto3.client("s3")
     successful = True
+    version = data["version"]
 
     for arch, arch_data in data["architectures"].items():
         logging.info(f" - Downloading minimal package for {arch_data['name']} ...")
+        filename = f"rti_connext_dds-{version}-min-{arch_data['name']}.zip"
         remote_file_path = PurePosixPath(
             storage_path,
             version,
-            f"rti_connext_dds-{version}-min-{arch_data['name']}.zip",
+            filename,
         )
+        download_dest = Path(f"/tmp/connext_downloads/{filename}")
+        download_dest.parent.mkdir(parents=True, exist_ok=True)
 
         try:
-            response = s3.get_object(Bucket=storage_url, Key=str(remote_file_path))
+            with open(download_dest, "wb") as f:
+                s3.download_fileobj(storage_url, str(remote_file_path), f)
         except ClientError:
             logging.error(
                 f"[Error] Could not find bundles for architecture {arch_data['name']}"
             )
             successful = False
             continue
-        compressed_file = io.BytesIO(response["Body"].read())
 
-        logging.info(f"  - Extracting necessary libs...")
+        logging.info("  - Extracting necessary libs...")
+
+        with zipfile.ZipFile(download_dest, "r") as zip_file:
+            connext_dds_dir = get_connext_dds_dir(zip_file)
+
+            if not connext_dds_dir:
```

Comment:
What happens on error? If we can enter the `for` statement, we should break from the `with` statement.

Reply:
Good catch!